pym/portage/dbapi/porttree.py

```diff
@@ -374,29 +416,29 @@ def findname2(self, mycpv, mytree=None, myrepo = None):
 		"""
 		if not mycpv:
 			return (None, 0)
-
-		if myrepo is not None:
-			mytree = self.treemap.get(myrepo)
-			if mytree is None:
-				return (None, 0)
-
+		
+		mytrees = []
 		mysplit = mycpv.split("/")
 		psplit = pkgsplit(mysplit[1])
 		if psplit is None or len(mysplit) != 2:
 			raise InvalidPackageName(mycpv)
 
+		cp = mysplit[0] + "/" + psplit[0]
+		if cp in self.better_cache:
+			for repo in self.better_cache[cp]:
+				if mytree and mytree != repo.location:
+					continue
+				elif (not myrepo) or \
+					(myrepo == repo.name) or \
+					(repo.aliases != None and myrepo in repo.aliases):
```

Comment:
Use `self.repoositories.get_location_for_name(myrepo)` here, it handles aliases.